### PR TITLE
docs: handling UTXO error in transaction re-submissions

### LINKED ISSUE #3009: Transaction Failure Due to Non-Existent UTXO After Retry Attempt
A user reported a transaction failure with the following error message:

```ts
FuelError: Transaction is not inserted. UTXO does not exist: 

0xbd65b960dc72e83d45ff11db256ced12ba4c1f8aaad05bfeeb83d72d1ac006b20003
```

The user indicated that the transaction initially failed due to a separate, unrelated error. Upon attempting to resubmit the transaction, the subsequent attempt failed with the error mentioned above.

This type of error suggests that this problem might be related to the [ResourcesCache](https://github.com/FuelLabs/fuels-ts/blob/c57f30ad888eb20aacf3aed0225131cb9a65a2b1/packages/account/src/providers/resource-cache.ts#L14)

__UPDATE__

After further validation, I believe the issue stems from repeated attempts to resubmit a transaction that failed during processing. Since the VM consumes transaction resources to cover the gas used up to the point of failure, attempting to resubmit the same transaction—now funded spent resources—results in the same error: `FuelError: Transaction is not inserted. UTXO does not exist: 0x...`

We are awaiting user confirmation on this.

### LINKED ISSUE #3049: Document Error for Failed Transaction Resubmission Due to Spent Resources
Related to https://github.com/FuelLabs/fuels-ts/issues/3009

We need to update our documentation to inform users that re-submitting a transaction request that failed during processing is likely to result in an error. 

This occurs because the VM will still consume the funded resources to cover the gas used up to the point when the transaction reverts. 

In such cases, the SDK will throw the following error:

```ts
FuelError: Transaction is not inserted. UTXO does not exist: {{utxoId}}
```

### PR DESCRIPTION
**Pull request by Torres-ssf**
- Closes #3049
- Closes #3009

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

File: apps/docs/src/guide/cookbook/resubmitting-failed-transactions.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Resubmitting Failed Transactions
+
+In certain scenarios, you might need to implement a solution to resubmit failed transactions to the Fuel Network. While this approach can be effective, there are important considerations to keep in mind.
```
Comment (arboleya) [inline apps/docs/src/guide/cookbook/resubmitting-failed-transactions.md:3]:
```suggestion
In certain scenarios, you might need to implement a solution to resubmit failed transactions to the Fuel Network. While this approach can be effective, there are important considerations to remember.
```

File: apps/docs/src/guide/cookbook/resubmitting-failed-transactions.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+While this can work, there is cautions to take when using this approach.
```
Comment (arboleya) [inline apps/docs/src/guide/cookbook/resubmitting-failed-transactions.md:5]:
```suggestion

```

File: apps/docs/src/guide/cookbook/resubmitting-failed-transactions.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+## Submission and Processing
+
+When submitting a transaction, it may fail either during submission or during processing.
```
Comment (arboleya) [inline apps/docs/src/guide/cookbook/resubmitting-failed-transactions.md:9]:
```suggestion
When submitting a transaction, you will first get a response.
```

File: apps/docs/src/guide/cookbook/resubmitting-failed-transactions.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+<<< @/../../docs-snippets/src/guide/cookbook/resubmitting-failed-transactions.test.ts#resubmitting-failed-transactions-1{ts:line-numbers}
+
+The `sendTransaction` method resolves without an error as soon as the transaction is successfully submitted and accepted by the network. However, this does not guarantee that the transaction has been processed; it only indicates that the transaction has been accepted and placed in a queue for processing.
```
Comment (arboleya) [inline apps/docs/src/guide/cookbook/resubmitting-failed-transactions.md:13]:
```suggestion
If the `sendTransaction` method resolves without an error, we know that the transaction was successfully submitted and accepted by the network. However, this does not guarantee that the transaction has been processed; it only indicates that the transaction has been accepted and placed in a queue for processing.
```

File: apps/docs/src/guide/cookbook/resubmitting-failed-transactions.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+To ensure that the transaction has been processed, you need to call `waitForResult`. This method will resolve (or reject with an error) only when the transaction has been fully processed.
```
Comment (arboleya) [inline apps/docs/src/guide/cookbook/resubmitting-failed-transactions.md:15]:
```suggestion
To determine whether the transaction has been processed, you must call `waitForResult`, which will either resolve (with the processed transaction) or reject with an error.
```

File: apps/docs/src/guide/cookbook/resubmitting-failed-transactions.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+<<< @/../../docs-snippets/src/guide/cookbook/resubmitting-failed-transactions.test.ts#resubmitting-failed-transactions-2{ts:line-numbers}
+
+In other words, if `sendTransaction` rejects with an error, it means that the transaction was not accepted by the network and was not processed. On the other hand, if the transaction fails during the call to `waitForResult`, it means that the transaction was accepted but was reverted during processing.
```
Comment (arboleya) [inline apps/docs/src/guide/cookbook/resubmitting-failed-transactions.md:19]:
```suggestion
In other words:
 - If `sendTransaction` is rejected with an error, the transaction was not accepted by the network and is not processed.
 - If `waitForResult` is rejected with an error, the transaction was accepted but reverted during processing.
```

File: apps/docs/src/guide/cookbook/resubmitting-failed-transactions.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+## Resources Spent When a Transaction Is Processed
+
+If a transaction is reverted during processing, the Fuel VM will still consume the funded resources to cover the gas used up to the point of failure. The remaining funds, after deducting the gas cost, will be added to a new UTXO (Unspent Transaction Output) addressed to the owner.
```
Comment (arboleya) [inline apps/docs/src/guide/cookbook/resubmitting-failed-transactions.md:23]:
```suggestion
If a transaction is reverted during processing, the Fuel VM will still consume the funded resources to cover the gas used up to the point of failure. After deducting the gas cost, the remaining funds will be added to a new UTXO (Unspent Transaction Output) addressed to the owner.
```

File: apps/docs/src/guide/cookbook/resubmitting-failed-transactions.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+<<< @/../../docs-snippets/src/guide/cookbook/resubmitting-failed-transactions.test.ts#resubmitting-failed-transactions-3{ts:line-numbers}
+
+The attempt from the above snippet will result in the error:
+
+```console
+FuelError: Transaction is not inserted. UTXO does not exist: {{utxoId}}
+```
+
+To safely retry a transaction that failed during processing, you should reassemble the transaction request from scratch and then resubmit it.
```
Comment (arboleya) [inline apps/docs/src/guide/cookbook/resubmitting-failed-transactions.md:35]:
```suggestion
To safely retry a transaction that failed during processing, you should reassemble the request from scratch and resubmit it.
```

**Review by arboleya — APPROVED** [review]
Good stuff! Left a couple of nits.

File: apps/docs-snippets/src/guide/cookbook/resubmitting-failed-transactions.test.ts [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+      const transactionRequest = await wallet.createTransfer(
+        recipient.address,
+        transferAmount,
+        baseAssetId
+      );
+
+      // Set the gasLimit to 0 to force revert with OutOfGas error
+      transactionRequest.gasLimit = bn(0);
+
+      // Transaction will successufully submitted
```
Comment (petertonysmith94) [inline apps/docs-snippets/src/guide/cookbook/resubmitting-failed-transactions.test.ts:62]:
```suggestion
      // Transaction will be successfully submitted
```

**Review by petertonysmith94 — DISMISSED** [review]

**Review by arboleya — APPROVED** [review]
These are excellent docs regarding this issue.

Thanks for the detailed explanations. 👌

**Review by petertonysmith94 — APPROVED** [review]

**Review by danielbate — APPROVED** [review]
🎯

**Review by maschad — APPROVED** [review]
👍🏾

**Pull request merged**